shell/common/shell.cc

```diff
@@ -822,20 +822,19 @@ void Shell::OnPlatformViewSetViewportMetrics(const ViewportMetrics& metrics) {
 
   // This is the formula Android uses.
   // https://android.googlesource.com/platform/frameworks/base/+/master/libs/hwui/renderthread/CacheManager.cpp#41
-  size_t max_bytes = metrics.physical_width * metrics.physical_height * 12 * 4;
-  task_runners_.GetRasterTaskRunner()->PostTask(
-      [rasterizer = rasterizer_->GetWeakPtr(), max_bytes] {
-        if (rasterizer) {
-          rasterizer->SetResourceCacheMaxBytes(max_bytes, false);
-        }
-      });
+  resource_cache_max_bytes_update_ =
+      metrics.physical_width * metrics.physical_height * 12 * 4;
 
   task_runners_.GetUITaskRunner()->PostTask(
       [engine = engine_->GetWeakPtr(), metrics]() {
         if (engine) {
           engine->SetViewportMetrics(metrics);
         }
       });
+
+  std::scoped_lock<std::mutex> lock(resize_mutex_);
```

Comment:
nit: consider creating a new scope for this

```
{
   lock;
   expeced_frame_size_ = ...;
}
```

When someone tries to add more functionality to this method, they will not hold the lock for longer than needed.